src/services/suggestionDiagnostics.ts

```diff
@@ -156,7 +156,29 @@ namespace ts {
     }
 
     function isPromiseHandler(node: Node): node is CallExpression {
-        return isCallExpression(node) && (hasPropertyAccessExpressionWithName(node, "then") || hasPropertyAccessExpressionWithName(node, "catch"));
+        return isCallExpression(node) && (
+            hasPropertyAccessExpressionWithName(node, "then") && hasZeroOrOneNonNullArguments(node) ||
+            hasPropertyAccessExpressionWithName(node, "catch"));
+    }
+
+    function hasZeroOrOneNonNullArguments(node: CallExpression) {

```

Comment:
I'm going to quibble with the name because it appears to return true for `then(null)`.  Maybe "secondArgumentIsNullOrAbsent"?

Also, it's not clear to me what having more than two arguments does - can I call `then(handler, undefined, undefined)`?  Are the extras just ignored?  What if the third argument is non-null?

Reply:
> I'm going to quibble with the name because it appears to return true for `then(null)`.

In `then(null)`, there are zero non-null arguments, so returning true is consistent with the name.

Arguments after the second have always been totally ignored (read: removed) by the refactor, but I’m guessing that was more accidental than intentional. It seems reasonable to disable the refactor to prevent code loss.